src/test/java/com/wikia/webdriver/elements/mercury/components/Search.java

```diff
@@ -1,57 +1,49 @@
+import com.wikia.webdriver.elements.mercury.pages.SearchResultsPage;
 import com.wikia.webdriver.pageobjectsfactory.pageobject.BasePageObject;
 
 import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.TimeoutException;
+import org.openqa.selenium.WebDriverException;
 import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
 import org.openqa.selenium.support.FindBy;
 
 public class Search extends BasePageObject {
 
-  @FindBy(css = ".wikia-search__container input")
+  @FindBy(css = ".wikia-search__container input.side-search__input")
   private WebElement searchInput;
 
   @FindBy(css = ".wikia-search__clear")
   private WebElement clearSearchButton;
 
-  private String searchResultClass = ".wikia-search__results li.mw-content a";
-  private By loadingSearchResultsIndicator = By.cssSelector(".wikia-search__results li.loading");
+  @FindBy(css = ".wikia-search__search-icon > svg > use[*|href*='#search']")
+  private WebElement inputFieldSearchIcon;
 
-  private Loading loading;
+  public static final int FOCUS_TIMEOUT_IN_SECONDS = 1;
+  public static final int SUGGESTIONS_TIMEOUT_IN_SECONDS = 1;
 
-  public Search() {
-    this.loading = new Loading(driver);
-  }
-
-  public Search seeNoSearchResults() {
-    PageObjectLogging.logInfo("Loading search results indicator is present");
-    wait.forElementPresent(loadingSearchResultsIndicator);
-
-    PageObjectLogging.logInfo("Loading search results indicator is not present");
-    wait.forElementNotPresent(loadingSearchResultsIndicator);
-
-    PageObjectLogging.logInfo("No search results are present");
-    wait.forElementNotPresent(By.cssSelector(searchResultClass));
-
-    return this;
-  }
+  private static final String searchSuggestionClass = ".wikia-search__suggestions li.mw-content a";
+  private static final String focusedSearchInput = ".wikia-search--focused .text-field-input";
 
-  public Search selectSearchSuggestion(int index) {
-    String oldUrl = driver.getCurrentUrl();
+  public String selectSearchSuggestion(int index) {
+    Loading loading = new Loading(driver);
+    String clickedLink;
 
     PageObjectLogging.logInfo("Select search suggestion no.: " + index);
-    WebElement searchResult = driver.findElements(By.cssSelector(searchResultClass)).get(index);
+
+    WebElement searchResult = driver.findElements(By.cssSelector(searchSuggestionClass)).get(index);
     wait.forElementClickable(searchResult);
+    clickedLink = searchResult.getAttribute("href");
+
     searchResult.click();
     loading.handleAsyncPageReload();
 
-    Assertion.assertFalse(oldUrl.equalsIgnoreCase(driver.getCurrentUrl()),
-                          "Navigation to selected search suggestion failed");
-    PageObjectLogging.logInfo("Successfully navigated to selected search suggestion");
-
-    return this;
+    return clickedLink;
```

Comment:
I'm thinking about returning the destination page object here. Thoughts on this idea and how to do it the cleanest?